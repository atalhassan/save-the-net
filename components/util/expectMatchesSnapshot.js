import renderer from "react-test-renderer";
import theme from "../../theme";
import { ThemeProvider } from "styled-components";
import React from "react";

/**
 * A wrapper around a generic snapshot test
 * 
 * @example
 * test("it renders correctly", () => {
 *   expectMatchesSnapshot(<Foo bar={"jazz"} />);
 * });
 */
function expectMatchesSnapshot(RenderedComponent) {
  if (!expect) {
    throw new Error(
      "expectMatchesSnapshot was used outside of a jest environment!"
    );
  }

  const tree = renderer
    .create(<ThemeProvider theme={theme}>{RenderedComponent}</ThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
}

export default expectMatchesSnapshot;
