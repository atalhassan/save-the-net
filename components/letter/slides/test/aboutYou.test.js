import AboutYou from "../aboutYou";
import expectMatchesSnapshot from "../../../util/expectMatchesSnapshot";

test("about you renders correctly without errors", () => {
  expectMatchesSnapshot(
    <AboutYou name={"George"} zipCode={83702} problems={[]} />
  );
});

test("about you renders correctly with errors", () => {
  expectMatchesSnapshot(
    <AboutYou name={"George"} zipCode={83702} problems={["name"]} />
  );
});
