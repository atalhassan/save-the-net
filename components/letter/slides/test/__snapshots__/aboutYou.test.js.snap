// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about you renders correctly with errors 1`] = `
<div>
  <h1>
    Tell us about you.
  </h1>
  <div
    className="inputs__TextGrouping-eera57-1 jqxINf"
  >
    <label
      className="inputs__Label-eera57-0 hyeqUX"
      htmlFor="name-input"
    >
      Name
       
    </label>
    <input
      className="inputs__Input-s1jrjbv6-0 cEReDj"
      id="name-input"
      name="name"
      onChange={undefined}
      placeholder="George Washington"
      type="text"
      value="George"
    />
  </div>
  <div
    className="inputs__TextGrouping-eera57-1 jqxINf"
  >
    <label
      className="inputs__Label-eera57-0 hyeqUX"
      htmlFor="zipCode-input"
    >
      Zip Code
       
    </label>
    <input
      className="inputs__Input-s1jrjbv6-0 igCRQP"
      id="zipCode-input"
      name="zipCode"
      onChange={undefined}
      placeholder="90210"
      type="number"
      value={83702}
    />
  </div>
  <button
    className="hukf77-0-buttons__Button-kzywbL iRrNVV"
    onClick={undefined}
  >
     Next 
  </button>
</div>
`;

exports[`about you renders correctly without errors 1`] = `
<div>
  <h1>
    Tell us about you.
  </h1>
  <div
    className="inputs__TextGrouping-eera57-1 jqxINf"
  >
    <label
      className="inputs__Label-eera57-0 hyeqUX"
      htmlFor="name-input"
    >
      Name
       
    </label>
    <input
      className="inputs__Input-s1jrjbv6-0 igCRQP"
      id="name-input"
      name="name"
      onChange={undefined}
      placeholder="George Washington"
      type="text"
      value="George"
    />
  </div>
  <div
    className="inputs__TextGrouping-eera57-1 jqxINf"
  >
    <label
      className="inputs__Label-eera57-0 hyeqUX"
      htmlFor="zipCode-input"
    >
      Zip Code
       
    </label>
    <input
      className="inputs__Input-s1jrjbv6-0 igCRQP"
      id="zipCode-input"
      name="zipCode"
      onChange={undefined}
      placeholder="90210"
      type="number"
      value={83702}
    />
  </div>
  <button
    className="hukf77-0-buttons__Button-kzywbL iRrNVV"
    onClick={undefined}
  >
     Next 
  </button>
</div>
`;
